webpack/assets/javascripts/react_app/routes/FiltersForm/FiltersForm.js

```diff
@@ -0,0 +1,261 @@
+import {
+  setAutocompleteSearchQuery,
+  updateDisability,
+  updateController,
+} from '../../components/AutoComplete/AutoCompleteActions';
+import './FilterForm.scss';
+import { SelectPermissions } from './SelectPermissions';
+import { SelectResourceType } from './SelectResourceType';
+import { SelectRole } from './SelectRole';
+import { EMPTY_RESOURCE_TYPE, SEARCH_ID } from './FiltersFormConstants';
+import { Taxonomies } from './Taxonomies';
+import { APIActions } from '../../redux/API';
+import { selectAutocompleteSearchQuery } from '../../components/AutoComplete/AutoCompleteSelectors';
+
+export const FiltersForm = ({ roleName, roleId, isNew, data, history }) => {
+  const [role, setRole] = useState(roleId);
+  const [type, setType] = useState(EMPTY_RESOURCE_TYPE);
+  const [chosenPermissions, setChosenPermissions] = useState([]);
+  const [isUnlimited, setIsUnlimited] = useState(!!data['unlimited?']);
+  const [isOverride, setIsOverride] = useState(!!data['override?']);
+  const [isGranular, setIsGranular] = useState(false);
+  const [chosenOrgs, setChosenOrgs] = useState(
+    data.organizations?.map(o => o.id) || []
+  );
+  const [chosenLocations, setChosenLocations] = useState(
+    data.locations?.map(l => l.id) || []
+  );
+  const {
+    show_organizations: showOrgs = false,
+    show_locations: showLocations = false,
+  } = type;
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (data.search)
+      dispatch(setAutocompleteSearchQuery(data.search, SEARCH_ID));
+    // workaround a bug - initial search query is ignored
+  }, [data.search, dispatch]);
+
+  const autocompleteQuery = useSelector(state =>
+    selectAutocompleteSearchQuery(state, SEARCH_ID)
+  );
+  const submit = async () => {
+    const params = {
+      filter: {
+        role_id: role,
+        search: autocompleteQuery,
+        unlimited: isUnlimited,
+        override: isOverride,
+        permission_ids: chosenPermissions,
+        organization_ids: chosenOrgs,
+        location_ids: chosenLocations,
+      },
+    };
+    const apiOptions = {
+      handleSuccess: () => {
+        history.push(`/filters?role_id=${roleId}`);
+      },
+      errorToast: ({ message }) => message,
+      params,
+    };
+    if (isNew) {
+      dispatch(
+        APIActions.post({
+          url: '/api/v2/filters',
+          key: 'POST_FILTERS_FORM',
+          successToast: () => __('Created role successfully'),
+          ...apiOptions,
+        })
+      );
+    } else {
+      dispatch(
+        APIActions.put({
+          url: `/api/v2/filters/${data.id}`,
+          key: 'PUT_FILTERS_FORM',
+          successToast: () => __('Edited role successfully'),
+          ...apiOptions,
+        })
+      );
+    }
+  };
+  return (
+    <Form className="filter-form">
+      {isNew ? (
+        <FormGroup label={__('Selected role')}>{roleName}</FormGroup>
+      ) : (
+        <SelectRole role={role} setRole={setRole} />
+      )}
+      <SelectResourceType
```

Comment:
Something weird happens when I click to edit a role, this is how the role looks like in the index page:
![Screenshot (95)](https://user-images.githubusercontent.com/26363699/179731841-ef3aaaa1-4ca1-4b21-abba-00f46dbf0ef8.png)

and then the resource type is set to Miscellaneous by default:
![Screenshot (96)](https://user-images.githubusercontent.com/26363699/179731941-69170aa6-05fc-4605-8926-28ecd5174377.png)

Reply:
Is there a network issue happening?

Reply:
it was a spaces issue in the resource type, fixed now